### ISSUE TITLE
Rewrite in Swift

### ISSUE THREAD

**Issue by AvdLee** [ashfurrow/UIView-BooleanAnimations#10]
If I'm right, loading pure Swift pods is better for performance. This repo is quite simple, a rewrite shouldn't take to much time. 

Curious about your thoughts!

**Comment by orta** [comment]
I'm not against it, but it's a full semver major breaking change, we use this in apps that don't have `use_frameworks!` enabled, and so won't be updating to it.

**Comment by AvdLee** [comment]
Good point. I'm thinking to start a new repo based on this, which suits better for both scenario's. Agree?

**Comment by orta** [comment]
For something this simple? I don't see a reason for splitting unless you're going to have a completely different API, it just adds extra choice for little gain. A Swift re-write is cool, but it's a 2.0, not a 1.1.

**Comment by AvdLee** [comment]
Alright, point is, we're trying to migrate as much frameworks as possible for us to Swift. I agree this one is simple, but in the end it would be great for us to move as much code as possible to Swift.

I'm happy to rewrite and create a pull request later on, which we can release in a 2.0.

**Comment by ashfurrow** [comment]
If Swift is your bag, I'd accept a pull request with a 2.0 rewrite. My only request is that you make it feel like an idiomatic Swift library, not just a port from Objective-C 😉 I figure if this will be a major semver change, we might as well make it the most awesome breaking change we can. Interop with Objective-C would still be important, but the library could benefit from use of some Swift-only features like default parameter values.
